lib/server.go

```diff
@@ -1822,6 +1856,49 @@ func (srv *Server) _addNewTxn(
 	return []*MsgDeSoTxn{txn}, nil
 }
 
+func (srv *Server) _addNewTxns(
+	pp *Peer, txns []*MsgDeSoTxn, rateLimit bool, verifySignatures bool) ([]*MsgDeSoTxn, error) {
+
+	if srv.ReadOnlyMode {
+		err := fmt.Errorf("Server._addNewTxnAndRelay: Not processing txn from peer %v "+
+			"because peer is in read-only mode: %v", pp, srv.ReadOnlyMode)
+		glog.V(1).Infof(err.Error())
+		return nil, err
+	}
+
+	if srv.blockchain.chainState() != SyncStateFullyCurrent {
+
+		err := fmt.Errorf("Server._addNewTxnAndRelay: Cannot process txn "+
+			"from peer %v while syncing: %v", pp, srv.blockchain.chainState())
+		glog.Error(err)
+		return nil, err
+	}
+
+	addedTxns := []*MsgDeSoTxn{}
+	for _, txn := range txns {
+		glog.V(1).Infof("Server._addNewTxnAndRelay: txn: %v, peer: %v", txn, pp)
+		// Always add the txn to the PoS mempool. This should always succeed if the txn
+		// addition into the PoW mempool succeeded above.
+		mempoolTxn := NewMempoolTransaction(txn, uint64(time.Now().UnixMicro()))
+		if err := srv.posMempool.AddTransaction(mempoolTxn, true /*verifySignatures*/); err != nil {
```

Comment:
Do we want to try to add the txn into the PoW mempool too? That said, it is fine to only support atomic txns after the PoS cutover